packages/proto-loader/bin/proto-loader-gen-types.ts

```diff
@@ -455,21 +447,46 @@ function generateMessageInterfaces(formatter: TextFormatter, messageType: Protob
 }
 
 function generateEnumInterface(formatter: TextFormatter, enumType: Protobuf.Enum, options: GeneratorOptions, nameOverride?: string) {
+  const {inputName, outputName} = useNameFmter(options);
+  const name = nameOverride ?? enumType.name;
   formatter.writeLine(`// Original file: ${(enumType.filename ?? 'null')?.replace(/\\/g, '/')}`);
   formatter.writeLine('');
   if (options.includeComments) {
     formatComment(formatter, enumType.comment);
   }
-  formatter.writeLine(`export enum ${nameOverride ?? enumType.name} {`);
+  formatter.writeLine(`export const ${name} = {`);
```

Comment:
This can easily cause conflicts. There is no guarantee that `name` is different from `inputName` or `outputName`, and in fact by default `inputName` is equal to `name`.

Also, how do you expect this object to be used by the user of the generated code?

Reply:
@murgatroid99 Because it's no longer using a TS enum, which is both a value and a type, we're allowed to have a type override for that value.

Here's an example output with the default input/outputNames and is valid TS:

```ts
// Original file: deps/gapic-showcase/schema/google/showcase/v1beta1/echo.proto

/**
 * A severity enum used to test enum capabilities in GAPIC surfaces
 */
export const Severity = {
  UNNECESSARY: 'UNNECESSARY',
  NECESSARY: 'NECESSARY',
  URGENT: 'URGENT',
  CRITICAL: 'CRITICAL',
} as const;

/**
 * A severity enum used to test enum capabilities in GAPIC surfaces
 */
export type Severity =
  | 'UNNECESSARY'
  | 0
  | 'NECESSARY'
  | 1
  | 'URGENT'
  | 2
  | 'CRITICAL'
  | 3

/**
 * A severity enum used to test enum capabilities in GAPIC surfaces
 */
export type Severity__Output = typeof Severity[keyof typeof Severity]
```

And a consuming message also is still valid:

```ts
// Original file: deps/gapic-showcase/schema/google/showcase/v1beta1/echo.proto

import type { Severity as _google_showcase_v1beta1_Severity, Severity__Output as _google_showcase_v1beta1_Severity__Output } from '../../../google/showcase/v1beta1/Severity';

/**
 * The response message for the Echo methods.
 */
export interface EchoResponse {
  /**
   * The content specified in the request.
   */
  'content'?: (string);
  /**
   * The severity specified in the request.
   */
  'severity'?: (_google_showcase_v1beta1_Severity);
}

/**
 * The response message for the Echo methods.
 */
export interface EchoResponse__Output {
  /**
   * The content specified in the request.
   */
  'content': (string);
  /**
   * The severity specified in the request.
   */
  'severity': (_google_showcase_v1beta1_Severity__Output);
}

```

The `const` is still there mostly for backwards compatibility. Right now the Enum name isn't modified at all by the `input` or `output` name formatting. So current users that import the old TS enum will now get the JS object with the values actually matching the configured `enum` type.